integration/client.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"time"
+
+	"github.com/m3db/m3db/interfaces/m3db"
+	"github.com/m3db/m3db/network/server/tchannelthrift/node"
+	"github.com/m3db/m3db/network/server/tchannelthrift/thrift/gen-go/rpc"
+	"github.com/m3db/m3db/services/m3dbnode/server"
+	xtime "github.com/m3db/m3db/x/time"
+
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/thrift"
+)
+
+func m3dbClient(address string, shardingScheme m3db.ShardScheme) m3db.Client {
+	return server.DefaultClient(address, shardingScheme)
+}
+
+func tchannelClient(address string) (*tchannel.Channel, rpc.TChanNode, error) {
+	channel, err := tchannel.NewChannel("integration-test", nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	endpoint := &thrift.ClientOptions{HostPort: address}
+	thriftClient := thrift.NewClient(channel, node.ChannelName, endpoint)
+	client := rpc.NewTChanNodeClient(thriftClient)
+	return channel, client, nil
+}
+
+// m3dbClientWriteBatch writes a data map using an m3db client.
+func m3dbClientWriteBatch(client m3db.Client, dm dataMap) error {
+	session, err := client.NewSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	for name, datapoints := range dm {
+		for _, dp := range datapoints {
+			if err := session.Write(name, dp.Timestamp, dp.Value, xtime.Second, nil); err != nil {
```

Comment:
Might want to do these in parallel with a worker pool as the client will try to batch the writes if they come in together quick enough.